msal4j-sdk/pom.xml

```diff
@@ -68,29 +68,41 @@
             <scope>test</scope>
         </dependency>
         <dependency>
-            <groupId>org.testng</groupId>
-            <artifactId>testng</artifactId>
-            <version>7.1.0</version>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-api</artifactId>
+            <version>5.9.2</version>
             <scope>test</scope>
         </dependency>
         <dependency>
-            <groupId>org.powermock</groupId>
-            <artifactId>powermock-module-testng</artifactId>
-            <version>2.0.0</version>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-params</artifactId>
+            <version>5.8.1</version>
             <scope>test</scope>
         </dependency>
         <dependency>
-            <groupId>org.powermock</groupId>
-            <artifactId>powermock-api-easymock</artifactId>
-            <version>2.0.0</version>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-engine</artifactId>
+            <version>5.9.2</version>
             <scope>test</scope>
         </dependency>
         <dependency>
-            <groupId>org.easymock</groupId>
-            <artifactId>easymock</artifactId>
-            <version>4.0.2</version>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-inline</artifactId>
+            <version>4.7.0</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-junit-jupiter</artifactId>
+            <version>4.7.0</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>net.bytebuddy</groupId>
```

Comment:
How is this runtime code generator used? Is this replacing mocking of private methods?

Reply:
Yes, it's needed to support the more advanced mocking options in `mockito-inline`. The `mockito-inline` package seems to require it (based on error messages when not using it), but it doesn't have it as a dependency for some reason so I needed to pull it in separately.

This `byte-buddy` and `mockito-inline` combo is [the same used by Azure SDK](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/identity/azure-identity/pom.xml#L101). It seems like [mockito-inline](https://mvnrepository.com/artifact/org.mockito/mockito-inline) is a side project of useful features (like private mocking and certain annotations) that will be in the main mockito package one day.

However, while looking at your comment I noticed I forgot to scope it to just be a 'test' dependency, and I fixed that in the latest commit.